### PR TITLE
[Incubating Plugin] Added various tests for compilationUnit override support

### PR DESCRIPTION
**Pull request by tasomaniac**

### CODE REVIEW

File: apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt [hunk 1]
```diff
@@ -351,6 +351,59 @@ class ConfigurationTests {
     }
   }
 
+  @Test
+  fun `sources can be overridden in compilationUnits`() {
+    withSimpleProject("""
+      apollo {
+        schemaPath = "com/some/other/schema.json"
```
Comment (martinbonnin) [inline apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt]:
```suggestion
        schemaFile = file("com/some/other/schema.json")
```

File: apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt [hunk 2]
```diff
@@ -351,6 +351,59 @@ class ConfigurationTests {
...
+        sourceFolder = "com/some/other"
```
Comment (martinbonnin) [inline apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt]: `sourceFolder` does not exist on the extension, it's only in services

File: apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt (hunk 1 above)
Comment (tasomaniac) [reply apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt]: Changing to this actually makes the test go green. I am not even sure if we have a bug to fix. 🤷‍♂️

File: apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt (hunk 2 above)
Comment (tasomaniac) [inline apollo-gradle-plugin-incubating/src/test/kotlin/com/apollographql/apollo/gradle/test/ConfigurationTests.kt]:
```suggestion
```

**Comment by tasomaniac** [comment]
@martinbonnin the issues I had (that led me work on these tests) were related to #1906 

Even putting `if (variantName == 'main')` and similar didn't work in our project last night :) They actually work now. That's why I'm not sure if we have a bug for this. 

Still, I think these tests can be merged since they're not bad.

**Pull request merged**